pkg/provider/provider.go

```diff
@@ -137,3 +161,44 @@ func convertToTypeSet(r *schema.Resource) *schema.Schema {
 		Elem: r,
 	}
 }
+
+// ValidateIAMVersion is a ValidateFunc for the "iam_version" field in the provider schema
+func ValidateIAMVersion(v interface{}, k string) ([]string, []error) {
+	// check that v is in iamVersionList
+	found := false
+	for _, version := range iamVersionList {
+		versn, ok := v.(string)
```

Comment:
This type assertion should be moved outside of the loop. Also could rename `versn` (which looks like a typo at first glance) to `versionInput` or some such. 

If we type assert `v` to `IAMVersion` instead of string, in line 175 we shouldn't need to cast `version` to string.

Reply:
I'm not sure that we can type assert to `IAMVersion` since terraform is going to give us a string.  We can _cast_ to IAMVersion once we've asserted that that the input is a string.

Unless I've missed something fundamental about golang types.